palace/utils/geodata.cpp

```diff
@@ -78,9 +83,18 @@ std::unique_ptr<mfem::ParMesh> ReadMesh(MPI_Comm comm, const IoData &iodata, boo
   const auto &refinement = iodata.model.refinement;
   const bool use_amr = refinement.max_it > 0;
   const bool use_mesh_partitioner = !use_amr || !refinement.nonconformal;
+  MPI_Comm node_comm;
+  if (!use_mesh_partitioner)
+  {
+    MPI_Comm_split_type(comm, MPI_COMM_TYPE_SHARED, Mpi::Rank(comm), MPI_INFO_NULL,
+                        &node_comm);
+  }
```

Comment:
To check I am getting this correctly, the idea is to split the communicator into N communicators, one for each node. Then within each communicator, only the root rank loads the serial, thereby avoiding the memory overloading issue of every rank on a node loading the mesh for preprocessing purposes.

Reply:
Correct. Just one process on each node does the serial mesh preprocessing operations to save memory overhead, and then broadcasts it to all other processes on the node.